src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -644,9 +684,101 @@ private boolean isAllowedApiRequest(HttpServletRequest request) {
+		boolean isCallAllowed = false;
+		//when using the substring method, the +9, +17, +x, the int is the 
+		//number of characters in the phrase for the indexOf method. The int 
+		//in this case is an offset.
+		String sectionString = url.substring(url.indexOf("sections/")+9, url.indexOf("/enrollments"));
+		String enrollmentTypeFromRequest = url.substring(url.indexOf("enrollment[type]=")+17, url.length());
+		M_log.debug("SECTION_STRING: " + sectionString);
+		M_log.debug("ENROLLMENT_TYPE: " + enrollmentTypeFromRequest);
+
+		//build api call
+		String sectionsApiCall = canvasURL + "/api/v1/sections/" + sectionString;
+		M_log.debug("SECTIONS API CALL: " + sectionsApiCall);
+
+		//string built, time to make call
+		String sessionId = request.getSession().getId();
+		String loggingApiWithSessionInfo = String.format("Canvas API request with Session Id \"%s\" for URL \"%s\"", sessionId,sectionsApiCall);
+		M_log.info(loggingApiWithSessionInfo);
+		HttpUriRequest clientRequest = null;
+
+		clientRequest = new HttpGet(sectionsApiCall);
+
+		BufferedReader rd = processApiCall(clientRequest);
+
+		String line = "";
+		StringBuilder sb = new StringBuilder();
+
+		try {
+			while ((line = rd.readLine()) != null) {
+				sb.append(line);
+			}
+		} catch (IOException e) {
+			M_log.error("Canvas API call did not complete successfully", e);
+		}
+		M_log.debug("RESPONSE TO isAddUserAllowed: " + sb.toString());
+
+		Integer courseIdFromRequest = null;
+		try{
+			//If JSON isn't properly formed catch exception
+			JSONObject sectionsCallResponse = new JSONObject( sb.toString() );
+			courseIdFromRequest = sectionsCallResponse.getInt("course_id");
+		}
+		catch(JSONException e){
+			M_log.error("JSONException found attempting to process sectionsCallResponse");
+			return false;
+		}
+		M_log.debug("COURSE ID FROM REQUEST: " + courseIdFromRequest);
+
+		HashMap<Integer, String> enrollmentsFound = (HashMap<Integer, String>) request.getSession().getAttribute("enrollments");
+		M_log.debug("SESSION ENROLLMENTS FOUND: " + request.getSession().getAttribute("enrollments"));
+
+		String enrollmentTypeFromSession = enrollmentsFound.get(courseIdFromRequest);
+		M_log.debug("ENROLLMENT FOUND: " + enrollmentTypeFromSession);
+
+		if(enrollmentTypeFromSession == null){
+			return false;
+		}
+
+		M_log.debug("ENROLLMENT FROM REQUEST: " + enrollmentTypeFromRequest);
+		
+		if(enrollmentsMap.containsKey(enrollmentTypeFromRequest) == false){
+			return false;
+		}
+		
+		int enrollmentValueFromRequest = enrollmentsMap.get(enrollmentTypeFromRequest);
+		int enrollmentValueFromSession = enrollmentsMap.get(enrollmentTypeFromSession);
+
+		M_log.debug("ENROLLMENT TYPE VALUE: " + enrollmentValueFromRequest);
+		M_log.debug("ENROLLMENT FOUND VALUE: " + enrollmentValueFromSession);
+
+		//Enrollment types are ordered by rank. If you are of the same or 
+		//higher rank, then you can add user with rank in request, otherwise 
+		//fail.
+		if(enrollmentValueFromSession == 2){
```

Comment:
Magic numbers :(.  How about some small methods that just do the comparisons?  If nothing else then the explanation and all the code referencing the values can be in the same section of the file rather than spread out across the file.  Small methods are fine if they clarify things.